src/event/event.rs

```diff
@@ -78,14 +78,32 @@ impl Default for Event {
 }
 
 impl Event {
-    /// Returns an [`Iterator`] for [`Attributes`]
-    pub fn attributes_iter<'a>(&'a self) -> impl Iterator<Item = (&'a str, AttributeValue<'a>)> {
+    /// Returns an [`Iterator`] for all the available [CloudEvents Context attributes](https://github.com/cloudevents/spec/blob/master/spec.md#context-attributes) and extensions.
+    /// Same as chaining [`Event::iter_attributes()`] and [`Event::iter_extensions()`]
+    pub fn iter(&self) -> impl Iterator<Item=(&str, AttributeValue)> {
```

Comment:
Was there a reason for not implementing the iterator trait?

Reply:
What do you mean? that I don't implement `IntoItererator`? The reason is that `IntoIterator` takes ownership of the `Event`: https://doc.rust-lang.org/std/iter/trait.IntoIterator.html

Reply:
No i see that the Iterator trait isn't what you need since you are chaining. Nvm xD